agent/stats/engine.go

```diff
@@ -697,17 +697,23 @@ func (engine *DockerStatsEngine) taskContainerMetricsUnsafe(taskArn string) ([]*
 				if !taskExistsInTaskStats {
 					return nil, fmt.Errorf("task not found")
 				}
-				networkStats, err := taskStatsMap.StatsQueue.GetNetworkStatsSet()
-				if err != nil {
-					seelog.Warnf("error getting network stats: %v, task: %v", err, taskArn)
+				if dockerContainer, err := engine.resolver.ResolveContainer(dockerID); err != nil {
+					seelog.Debugf("Could not map container ID to container, container: %s, err: %s", dockerID, err)
```

Comment:
if there's an error with `engine.resolver.ResolveContainer(dockerID)` we log, otherwise we get our networkStats.  This potentially means that we would lose networkStats if the ResolveContainer call fails for any reason.

Is it preferable to wrongly ADD the pause container stats if we fail the ResolveContainer call, rather than fail and swallow a valid container's stats?

```
networkStats, err := taskStatsMap.StatsQueue.GetNetworkStatsSet()
if dockerContainer, err := engine.resolver.ResolveContainer(dockerID); err != nil {
    seelog.Debugf("Could not map container ID to container, container: %s, err: %s", dockerID, err)
    // do we still want to send network stats here, as we aren't sure which container this is?
    containerMetric.NetworkStatsSet = networkStats
} else {
    if dockerContainer.Container.Type != apicontainer.ContainerCNIPause {
        containerMetric.NetworkStatsSet = networkStats
    }
}
```

Reply:
I think its ok not to show metrics for a container that agent does not account in its state. 

We do something similar while resolving task as well, and ignore stats if its not found: 
https://github.com/aws/amazon-ecs-agent/blob/2874131a5f3d1c0908e0f2ebad62e48b8ff5850f/agent/stats/engine.go#L681

and few other instances too

https://github.com/aws/amazon-ecs-agent/blob/2874131a5f3d1c0908e0f2ebad62e48b8ff5850f/agent/stats/engine.go#L533